src/utils/ejson.ts

```diff
@@ -0,0 +1,46 @@
+import { EJSON } from 'bson';
+import type { Document } from 'bson';
+
+const isObject = (value: unknown) =>
+  value !== null && typeof value === 'object';
+
+function simplifyEJSON(documents: Document[] | Document): Document {
+  if (!isObject(documents)) return documents;
+
+  if (Array.isArray(documents)) {
+    return documents.map((item) =>
+      isObject(item) ? simplifyEJSON(item) : item
+    );
+  }
+
+  // UUIDs might be represented as {"$uuid": <canonical textual representation of a UUID>} in EJSON
+  // Binary subtypes 3 or 4 are used to represent UUIDs in BSON
+  // But, parsers MUST interpret the $uuid key as BSON Binary subtype 4
+  // For this reason, we are applying this representation for subtype 4 only
+  // see https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#special-rules-for-parsing-uuid-fields
+  if (
+    Object.prototype.hasOwnProperty.call(documents, '$binary') &&
```

Comment:
Do we need to check `hasOwnProperty` here? Isn't it enough to check `documents.$binary?.something`?